app/src/main/java/com/kks/nimblesurveyjetpackcompose/repo/home/HomeRepoImpl.kt

```diff
@@ -0,0 +1,52 @@
+package com.kks.nimblesurveyjetpackcompose.repo.home
+
+import com.kks.nimblesurveyjetpackcompose.model.ResourceState
+import com.kks.nimblesurveyjetpackcompose.model.response.SurveyResponse
+import com.kks.nimblesurveyjetpackcompose.model.response.UserResponse
+import com.kks.nimblesurveyjetpackcompose.network.ApiInterface
+import com.kks.nimblesurveyjetpackcompose.util.extensions.SUCCESS_WITH_NULL_ERROR
+import com.kks.nimblesurveyjetpackcompose.util.extensions.safeApiCall
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+import javax.inject.Inject
+
+class HomeRepoImpl @Inject constructor(
+    private val apiInterface: ApiInterface
+) : HomeRepo {
+
+    override fun fetchSurveyList(
+        pageNumber: Int,
+        pageSize: Int
+    ): Flow<ResourceState<List<SurveyResponse>>> = flow {
+        val apiResult = safeApiCall(Dispatchers.IO) { apiInterface.getSurveyList(pageNumber, pageSize) }
+        when (apiResult) {
+            is ResourceState.Success -> {
+                apiResult.data?.data?.let {
+                    emit(ResourceState.Success(it))
+                } ?: emit(ResourceState.Success(listOf()))
```

Comment:
```suggestion
                } ?: emit(ResourceState.Success(emptyList()))
```
Or better if we emit some error.

Reply:
I've made the change in [Home Screen PR](https://github.com/kaungkhantsoe/NimbleSurveyJetpackCompose/pull/74)